### PR TITLE
feat(page): working shortcut toggle menu item

### PR DESCRIPTION
**Pull request by shanberg**
[Demo](https://shanberg.github.io/athens/)

Working add/remove shortcut button in page menu

### CODE REVIEW

File: src/cljs/athens/events.cljs [hunk 1]
```diff
@@ -284,6 +284,22 @@
...
 
 
+(reg-event-fx
+  :page/make-shortcut
+  (fn [_ [_ uid]]
+    (let [sidebar-ents (d/q '[:find ?e
+                              :where
+                              [?e :page/sidebar _]]
+                            @db/dsdb)]
+      {:transact! [{:block/uid uid :page/sidebar (count sidebar-ents)}]})))
```
Comment (tangjeff0) [inline src/cljs/athens/events.cljs:294]: I would use `count` in the datalog query. It likely is optimized in someway

```
(d/q '[:find (count ?e) .
 :where
 [?e :page/sidebar _]]
 @db/dsdb)
```

File: src/cljs/athens/events.cljs [hunk 2]
```diff
@@ -284,6 +284,22 @@
...
+
+
+(reg-event-fx
+  :page/unmake-shortcut
+  (fn [_ [_ uid]]
+    {:transact! [[:db/retract [:block/uid uid] :page/sidebar]]}))
```
Comment (tangjeff0) [inline src/cljs/athens/events.cljs:300]: Honestly really impressed you used `:block/uid` as the identifier for these two queries. I didn't know that was possible / hand't thought to do that.

Comment (tangjeff0) [reply src/cljs/athens/events.cljs:300]: Thoughts on `:page/add-shortcut` and `:page/remove-shortcut`? That's the label of the buttons

File: src/cljs/athens/views/dropdown.cljs [hunk 3]
```diff
@@ -144,12 +145,19 @@
 
 
 (defn page-menu-component
-  [{:keys [style]}]
+  [{:keys [style uid has-shortcut?]}]
```
Comment (tangjeff0) [inline src/cljs/athens/views/dropdown.cljs:148]: What about `is-shortcut?`

File: src/cljs/athens/views/node_page.cljs [hunk 4]
```diff
@@ -218,7 +223,11 @@
              :label [:> mui-icons/ExpandMore]
              :style page-menu-toggle-style}]
     (when @show-page-menu?
-      [page-menu-component {:style {:position "fixed" :left (str (:x @page-menu-position) "px") :top (str (:y @page-menu-position) "px")}}])
+      [page-menu-component {:style {:position "fixed"
```
Comment (tangjeff0) [inline src/cljs/athens/views/node_page.cljs:226]: Where were we on the interface to our components? Passing a style map 2-3 levels here.

Comment (shanberg) [reply src/cljs/athens/views/node_page.cljs:226]: Here: #278

**Comment by tangjeff0** [comment]
Going to work off this branch

**Comment by shanberg** [comment]
Obsoleted by #289

**Pull request closed without merging**